### PR TITLE
Add formatted printing directly into std.Buffer

### PR DESCRIPTION
**Pull request by vegecode**

### CODE REVIEW

**Comment by daurnimator** [comment]
> We used to have this, and removed it in favor of One Way To Do Things (create a stream from the buffer).
> 
> However, upon reconsidering, I think this API shortcut is welcome here.

I disagree. `std.Buffer` is unfortunately used in several places that don't make sense (probably because of the name?). e.g. there is no reason for `readUntilDelimiterAlloc` to use a `std.Buffer`.
Perhaps we should rename it to `ZeroTerminatedArrayList` or similar to make its differentiating property more obvious?

**Comment by vegecode** [comment]
It was totally the name that made me pick it originally. I ended up going with a different solution.

File: lib/std/buffer.zig [hunk 1]
```diff
@@ -147,6 +147,16 @@ pub const Buffer = struct {
         try self.resize(m.len);
         mem.copy(u8, self.list.toSlice(), m);
     }
+
+    pub fn print(self: *Buffer, comptime fmt: []const u8, args: var) !void {
+        try std.fmt.format(
+            self,
+            @TypeOf(Buffer.append).ReturnType.ErrorSet,
```
[the remark this answers is not in the document]
Comment (vegecode) [reply lib/std/buffer.zig]: Done. Had to add ".ErrorSet" to the end of your suggestion.

**Comment by vegecode** [comment]
What should I do about the windows build timing out?

Also @daurnimator , the Buffer has a convenient api for string operations such as:

```
startsWith
endsWith
eql
```

For someone browsing the standard library, it seems to be the obvious choice.

Is the ArrayList preferred for holding variable length u8's?

**Comment by daurnimator** [comment]
> Also @daurnimator , the Buffer has a convenient api for string operations such as:
> 
> ```
> startsWith
> endsWith
> eql
> ```
> 
> For someone browsing the standard library, it seems to be the obvious choice.

Adding in https://github.com/ziglang/zig/pull/4405
 
> Is the ArrayList preferred for holding variable length u8's?

Yes. Or `LinearFifo` if you ever need to delete data from the start.

**Comment by andrewrk** [comment]
Thanks!

**Pull request closed without merging**